File: pset1/cash/cash.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>

int main(void)
{
    // Prompt user for amount of change owed
    float change;
    do
    {
        change = get_float("Change owed: ");
    }
    while (change < 0);
    int cents = round(change * 100);

    // Calculate min number of coins to make change
    int count = 0;
    int coins[4] = {25, 10, 5, 1};

    // If still change to give check quarters
    int i = 0;
    while (cents > 0)
    {
        count += cents / coins[i];
        cents = cents % coins[i];
        i++;
    }

    // Print min number of coins
    printf("%i\n", count);
}

File: week6/pset6/cash/cash.py
# Port of cash.c
from cs50 import get_float

# Get non-negative change owed from user
change = get_float("Change owed: ")
while change < 0:
    change = get_float("Change owed: ")

# Possible coin denominations (in cents)
coins = [25, 10, 5, 1]

# Initialize coin list index and coin count
coin = 0
count = 0

# Convert change to cents
change *= 100

# Loop until no change remaining
while change > 0:
    count += int(change / coins[coin])
    change = change % coins[coin]
    coin += 1

# Print min number of coins needed
print(f"{count}")